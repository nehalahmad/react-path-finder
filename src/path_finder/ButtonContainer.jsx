import React from "react";
import PropTypes from "prop-types";
import { Button } from "react-bootstrap";
import { RESET_TEXT } from "../config/AppConstants";

/**
 * @description: A wrapper container for form buttons
 */
const ButtonContainer = props => {
  const { isLoader, submitBtnText } = props;

  return (
    <div className="button-container text-right">
      <Button type="submit" variant="primary" disabled={isLoader}>
        {submitBtnText}
      </Button>
      <Button type="reset" variant="reset">
        {RESET_TEXT}
      </Button>
    </div>
  );
};

// validate prop types
ButtonContainer.propTypes = {
  isLoader: PropTypes.bool,
  submitBtnText: PropTypes.string
};

export default ButtonContainer;
