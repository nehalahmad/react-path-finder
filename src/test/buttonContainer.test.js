import React from 'react';
import ButtonContainer from './../components/form/components/buttonContainer';
import {shallow} from 'enzyme';

describe("<ButtonContainer />", () => {
  it("button container renders with 'Submit' text", () => {
    const buttonContainerObj = shallow(
      <ButtonContainer submitBtnText="Submit" />
    );
    expect(buttonContainerObj).toMatchSnapshot();
  });

  it("button container renders with 'Re-submit' text", () => {
    const buttonContainerObj = shallow(
      <ButtonContainer submitBtnText="Re-submit" />
    );
    expect(buttonContainerObj).toMatchSnapshot();
  });
});