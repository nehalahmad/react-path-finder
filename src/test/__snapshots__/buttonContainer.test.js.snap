// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonContainer /> button container renders with 'Re-submit' text 1`] = `ShallowWrapper {}`;

exports[`<ButtonContainer /> button container renders with 'Submit' text 1`] = `ShallowWrapper {}`;
